Source/WebGPU/WebGPU/Queue.h

```diff
@@ -123,7 +124,7 @@ private PUBLIC_IN_WEBGPU_SWIFT:
     HashMap<uint64_t, OnSubmittedWorkScheduledCallbacks, DefaultHash<uint64_t>, WTF::UnsignedWithZeroKeyHashTraits<uint64_t>> m_onSubmittedWorkScheduledCallbacks;
     using OnSubmittedWorkDoneCallbacks = Vector<WTF::Function<void(WGPUQueueWorkDoneStatus)>>;
     HashMap<uint64_t, OnSubmittedWorkDoneCallbacks, DefaultHash<uint64_t>, WTF::UnsignedWithZeroKeyHashTraits<uint64_t>> m_onSubmittedWorkDoneCallbacks;
-    HashMap<uint32_t, RefPtr<CommandEncoder>, DefaultHash<uint32_t>, WTF::UnsignedWithZeroKeyHashTraits<uint32_t>> m_retainedEncoders;
+    NSMutableDictionary<NSNumber*, NSMutableSet<id<MTLCounterSampleBuffer>>*>* m_retainedCounterSampleBuffers;
```

Comment:
To ensure no retain cycle from CommandEncoder -> MTLCommandBuffer -> CommandEncoder, I retain updated the PR to only retain the MTLCounterSampleBuffer instead